imap_processing/hi/utils.py

```diff
@@ -25,3 +32,94 @@ def sensor(self) -> str:
             "45sensor" or "90sensor".
         """
         return self.name[1:3] + "sensor"
+
+
+def full_dataarray(
+    name: str,
+    attrs: dict,
+    coords: Optional[dict[str, xr.DataArray]] = None,
+    shape: Optional[Union[int, Sequence[int]]] = None,
+) -> xr.DataArray:
+    """
+    Generate an empty xarray.DataArray with appropriate attributes.
+
+    Data in DataArray are filled with FILLVAL defined in attributes
+    retrieved from ATTR_MGR with shape matching coordinates defined by
+    dims or overridden by optional `shape` input.
+
+    Parameters
+    ----------
+    name : str
+        Variable name.
+    attrs : dict
+        CDF variable attributes. Usually retrieved from ImapCdfAttributes.
+    coords : dict
+        Coordinate variables for the Dataset.
+    shape : int or tuple
+        Shape of ndarray data array to instantiate in the xarray.DataArray.
+
+    Returns
+    -------
+    data_array : xarray.DataArray
+        Meeting input specifications.
+    """
+    _attrs = attrs.copy()
```

Comment:
I think that you meant to use _attrs in the rest of the function after this line.

Reply:
Good catch!